parser/src/main/java/com/linkedin/avroutil1/parser/avsc/AvroParseContext.java

```diff
@@ -71,19 +71,37 @@ public void resolveReferences() {
         //TODO - add context-level issues for dups
 
         //resolve any unresolved references in individual file results from other files
-        externalReferences = new HashMap<>(1);
+        externalReferences = new ArrayList<>();
         for (AvscParseResult singleFile : individualResults) {
-            Map<String, List<SchemaOrRef>> externalRefs = singleFile.getExternalReferences();
-            externalRefs.forEach((fqcn, refs) -> {
-                AvscParseResult otherFile = knownNamedSchemas.get(fqcn);
-                if (otherFile == null) {
+            List<SchemaOrRef> externalRefs = singleFile.getExternalReferences();
+            for (SchemaOrRef ref : externalRefs) {
+                String simpleName = ref.getRef();
+                AvscParseResult simpleNameResolution = knownNamedSchemas.get(simpleName);
+                AvscParseResult inheritedNameResolution = null;
+
+                String inheritedName = ref.getInheritedName();
+                if (inheritedName != null) {
+                    inheritedNameResolution = knownNamedSchemas.get(inheritedName);
+                }
+
+                // The namespace may be inherited from the parent schema's context or may already be defined in the
+                // name. There may be multiple resolutions for a simple name (either from the null namespace or from
+                // the inherited namespace).
+                if (inheritedNameResolution != null) {
+                    ref.setResolvedTo(inheritedNameResolution.getDefinedNamedSchemas().get(inheritedName));
+                    if (simpleNameResolution != null) {
+                        String msg =
```

Comment:
AvscParseResult has "issues". this should be added to "issues" for the schema with the ref in question

Reply:
Currently, there is no ability to add new issues to an existing `AvscParseResult`. Do you see there being any side-effects with modifying the existing `issues` in `AvscParseResult`?

Reply:
should be fine to add issues. we dont claim the model is thread-safe and this happens during parsing, so the results have not been "finalized" yet